tests/test_error_message_builder.py

```diff
@@ -0,0 +1,220 @@
+                  .message("Message with {{parameter_name1}} and "
+                           "{{parameter_name2}}.", "value1", "value2"))
+    assert message == "E-ERJ-TEST-1: Message with 'value1' and 'value2'."
+
+
+def test_message_inline_unquoted_parameter_without_name():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("Message with {{|uq}}.", "value"))
+    assert message == "E-ERJ-TEST-1: Message with value."
+
+
+def test_message_inline_unquoted_parameter_without_value():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("Message with {{parameter_name|uq}}."))
+    assert message == "E-ERJ-TEST-1: Message with <null>."
+
+
+def test_message_inline_single_unquoted_parameter_none_value():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("Message with {{parameter_name|uq}}.", None))
+    assert message == "E-ERJ-TEST-1: Message with <null>."
+
+
+def test_message_inline_multiple_unquoted_parameter_none_value():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("Message with {{parameter_name|uq}} and "
+                           "{{parameter_name2}}.", None, None))
+    assert message == "E-ERJ-TEST-1: Message with <null> and <null>."
+
+
+def test_message_inline_multiple_unquoted_parameter_none_parameters():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("Message with {{parameter_name|uq}} and "
+                           "{{parameter_name2}}.")
+                  .parameter("parameter_name", None)
+                  .parameter("parameter_name2", None))
+    assert message == "E-ERJ-TEST-1: Message with <null> and <null>."
+
+
+def test_message_inline_unquoted_parameter_with_group_reference_char():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("test {{my_placeholder|uq}}")
+                  .parameter("my_placeholder", "$2"))
+    assert message == "E-ERJ-TEST-1: test $2"
+
+
+def test_message_inline_and_outline_order():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("Message with {{parameter_name1}}", "value1")
+                  .message(" {{parameter_name2}}.", "value2"))
+    assert message == "E-ERJ-TEST-1: Message with 'value1' 'value2'."
+
+
+def test_message_inline_and_outline_unquoted_parameters():
+    message = str(ErrorMessageBuilder("E-ERJ-TEST-1")
+                  .message("Message with {{parameter_name1|uq}}", "value1")
+                  .message(" {{parameter_name2|uq}}.", "value2"))
+    assert message == "E-ERJ-TEST-1: Message with value1 value2."
+
+
+class TestErrorCode(unittest.TestCase):
+    def test_invalid_error_code_format(self):
+        invalid_error_code_list = [
+            "A-EXA-100",
+            "F1-EXA-100",
+            "F-1EXA-100",
+            "F-EXA-EXA-F1"
+            "F-EXA-EXA",
+            "F-100"
+        ]
```

Comment:
The error-codes document in the confluence page is not so clear to me, I'm not so sure if these codes are invalid.